services/QuillLMS/app/models/activity.rb

```diff
@@ -313,6 +313,18 @@ def serialize_with_topics_and_publication_date
     serializable_hash.merge({topics: topics&.map(&:genealogy), publication_date: publication_date})
   end
 
+  def update_questions_flag_status_if_necessary!
+    return unless flags.include?(:production)
+
+    question_uids = data["questions"].map{|q| q["key"]}
+    questions = Question.where("uid in (?)", question_uids)
```

Comment:
This is fine as-is, but it feels like these could be two methods on the model that could be useful in other places:  (Also note the `&` null-safe syntax so that this won't crash if somehow `data` doesn't have a "questions" key.)
```
def question_uids = data["questions"]&.map{|q| q["key"]}

def questions = Question.where(uid: question_uids)
```

Reply:
I agree! I felt like it would be nice to be able to just call `activity.questions` but didn't want to go too into the weeds.

Reply:
updated to use the `?`